scanner/updater/export.go

```diff
@@ -0,0 +1,106 @@
+package updater
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/klauspost/compress/zstd"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/libvuln/jsonblob"
+	"github.com/quay/claircore/libvuln/updates"
+	_ "github.com/quay/claircore/updater/defaults"
+	"github.com/quay/zlog"
+	"github.com/stackrox/stackrox/scanner/v4/updater/manual"
+	"golang.org/x/time/rate"
+)
+
+// Export is responsible for triggering the updaters to download Common Vulnerabilities and Exposures (CVEs) data
+// and then outputting the result as a zstd-compressed file with .ztd extension
+func Export(ctx context.Context, outputDir string) error {
+	var outOfTree []driver.Updater
+
+	// Append updater sets directly to the outOfTree
+	appendUpdaterSet := func(updaterSet driver.UpdaterSet, err error) {
+		if err != nil {
+			zlog.Error(ctx).Err(err).Send()
+			return
+		}
+		outOfTree = append(outOfTree, updaterSet.Updaters()...)
+	}
+
+	appendUpdaterSet(manual.UpdaterSet(ctx, nil))
+
+	// create temp folder
+	err := os.Mkdir(outputDir, 0700)
```

Comment:
I think maybe https://pkg.go.dev/os#MkdirAll in case the directory already exists